service/create/launch_configuration.go

```diff
@@ -0,0 +1,146 @@
+package create
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/aws-operator/resources"
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/certificatetpr"
+	cloudconfig "github.com/giantswarm/k8scloudconfig"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+	awsresources "github.com/giantswarm/aws-operator/resources/aws"
+)
+
+type launchConfigurationInput struct {
+	name                string
+	clients             awsutil.Clients
+	cluster             awstpr.CustomObject
+	tlsAssets           *certificatetpr.CompactTLSAssets
+	bucket              resources.Resource
+	securityGroup       resources.ResourceWithID
+	subnet              *awsresources.Subnet
+	keypairName         string
+	instanceProfileName string
+	prefix              string
+}
+
+func (s *Service) createLaunchConfiguration(input launchConfigurationInput) (bool, error) {
+	var (
+		extension    cloudconfig.Extension
+		imageID      string
+		instanceType string
+	)
+
+	switch input.prefix {
+	case prefixMaster:
+		extension = NewMasterCloudConfigExtension(input.cluster.Spec, input.tlsAssets)
+
+		// TODO Check only a single master node is provided.
+		imageID = input.cluster.Spec.AWS.Masters[0].ImageID
+		instanceType = input.cluster.Spec.AWS.Masters[0].InstanceType
+	case prefixWorker:
+		extension = NewWorkerCloudConfigExtension(input.cluster.Spec, input.tlsAssets)
+
+		// TODO Until multiple worker instance types supported check only a single
```

Comment:
We don't want to add the autoscaling group concept to clustertpr as its AWS specific. So we need to parse the set of instance types for the cluster TPO. Until we support multiple ASGs we should error if there are different instance types in a cluster. Raised #304 for this